### DISCUSSION TITLE
CV-CUDA Release v0.4.0

### DISCUSSION THREAD

**Discussion by milesp-nvidia** [CVCUDA/CV-CUDA#108] [Announcements]
CV-CUDA 0.4.0 is a major release of the library providing multiple new operators, Jetson Orin support, and updated API documentation.

## Release Highlights

CV-CUDA v0.4.0 includes the following key features:

- 14 new image processing and computer vision operators
  - Advanced Color Format Conversion
  - Brightness_Contrast
  - Color_Twist
  - FindContours
  - GaussianNoise
  - Histogram
  - Histogram Equalizer
  - Inpainting
  - MinAreaRect
  - MinMaxLoc
  - Morphology (Open, Close)
  - On-screen display (Polyline, Point, Line, Text, Rotated Rectangle, Segmented Mask)
  - RandomResizedCrop
  - SIFT
- Updated sample application
  - Streamed Triton-based Video Segmentation Sample using CV-CUDA and VPF (Video Processing Framework) optimized for performance with video decode/encode on server-side
- Added Jetson Orin support for core library
- Updated API documentation

## Compatibility

CV-CUDA has been tested on the following compute stack:

- GPU Compute Capability: 7+.x
- Ubuntu x86_64: 20.04, 22.04
- CUDA Toolkit: 11.7+ (11.2+ for library build and run)
- GCC: 11.0+ (9.0 and 10.0 for APIs, with pre-built binary and run)
- Python: 3.7, 3.8, 3.10

Refer to documentation of the sample applications for dependencies.

## Known Issues/Limitations

- Samples fails for encoding surfaces on T4 with CUDA 11.8 and display driver 520. Suggested workaround is to upgrade to a newer driver 525+.
- For GCC versions lower than 11.0, C++17 support needs to be enabled when compiling CV-CUDA.

## License

CV-CUDA operates under the [Apache 2.0](https://github.com/CVCUDA/CV-CUDA/blob/main/LICENSE.md) license.

## Resources

1. [CV-CUDA GitHub](https://github.com/CVCUDA/CV-CUDA)
2. [CV-CUDA Increasing Throughput and Reducing Costs for AI-Based Computer Vision with CV-CUDA](https://developer.nvidia.com/blog/increasing-throughput-and-reducing-costs-for-computer-vision-with-cv-cuda/)
3. [NVIDIA Announces Microsoft, Tencent, Baidu Adopting CV-CUDA for Computer Vision AI](https://blogs.nvidia.com/blog/2023/03/21/cv-cuda-ai-computer-vision/)
4. [CV-CUDA helps Tencent Cloud audio and video PaaS platform achieve full-process GPU acceleration for video enhancement AI](https://developer.nvidia.com/zh-cn/blog/cv-cuda-high-performance-image-processing/)

## Acknowledgements

CV-CUDA is developed jointly by NVIDIA and the ByteDance Machine Learning team.

<hr /><em>This discussion was created from the release <a href='https://github.com/CVCUDA/CV-CUDA/releases/tag/v0.4.0-beta'>CV-CUDA Release v0.4.0</a>.</em>

**Comment by xiejichao** [comment]
where is nvcv-python3.10
